### DISCUSSION TITLE
About the divide-by generated clock domain

### DISCUSSION THREAD

**Discussion by wswslzp** [SpinalHDL/SpinalHDL#680] [General]
It's found that the SpinalHDL support generating a new slower clock domain from a source clock, by `ClockDomain.newClockDomainSlowedBy(factor)`. 

However, the source code shows that the clock divider doesn't produce a new clock signal as a normal clock divider does, but a clock enable signal that ticks every N cycle. Though it works in most cases, it still confused me. 

I'm curious about the reason behind this design choice. Is it a kind of resource-saving approach on FPGA?

**Comment by Dolu1990** [comment]
> , but a clock enable signal that ticks every N cycle.

Yes, the idea was to keep  thing as inferable as possible and not using a clock-tree for it. Creating a clock by dividing it with a counter would very likely work too, but it is a less "pure digital" aproach. Still it could be added as a alternative.

**Reply by MarekPikula** [reply]
Actually, in some circumstances direct division is needed. For example, recently I needed to connect a slow clock to some BlackBox and I couldn't do it with the current implementation of slow clock domain because the BlackBox didn't support clockEnable. I ended up using vendor's IP block for clock division (which is probably the better option either way) but still I'd appreciate an option to use direct clock division  in the slow clock domain on the testing stage.

**Comment by wswslzp** [comment]
Thanks for the reply!
